### ISSUE TITLE
PEP 561 type information omitted

### ISSUE THREAD

**Issue by JacobHenner** [noirello/bonsai#74]
bonsai appears to be missing [PEP 561](https://peps.python.org/pep-0561/) type information, even though bonsai itself has type annotations. This is preventing mypy from type checking when the library is used in other projects:

```
error: Skipping analyzing "bonsai": module is installed, but missing library stubs or py.typed marker
note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

**Comment by noirello** [comment]
I started a [new branch](https://github.com/noirello/bonsai/tree/py-typed), added the `py.typed` file. The module indeed has some type annotations, and I'm trying to improve it, but it's unlikely that it's every going to be perfect. A lots of dynamic feature is going on under the hood.

**Comment by noirello** [comment]
The `py.typed` file is included in the latest 1.5.1 release.

But like I said, the type annotations aren't the best of the module. Made a couple of bad design decisions over the years in regard of typings, that could only be addressed in a new, major release version. Not sure that's ever going to happen, though.
